src/Components/src/Microsoft.AspNetCore.Components/Rendering/ComponentState.cs

```diff
@@ -139,16 +140,16 @@ public void SetDirectParameters(ParameterCollection parameters)
                 parameters = parameters.WithCascadingParameters(_cascadingParameters);
             }
 
-            Component.SetParameters(parameters);
+            _renderer.AddToPendingTasks(Component.SetParametersAsync(parameters));
         }
 
-        public void NotifyCascadingValueChanged()
+        public async Task NotifyCascadingValueChanged()
```

Comment:
Do we definitely need to make this `async`? The recipient of a changed-cascading-value can choose to do some async rendering, but the upstream caller doesn't have to wait for it. I would have thought we could make this equivalent to setting normal (non-cascaded) params, in that it's set-and-forget for the supplier, and the recipient chooses to coordinate its async task with the renderer if it wants.

I'd be keen *not* to introduce `async` here if we can, because it makes the upstream CascadingValue highly allocatey (awaiting a `Task.WhenAll`, etc.), and it doesn't seem like it buys us anything.

Reply:
I've removed all the asyncness